### PR TITLE
Add C# send HTTP request

### PR DESCRIPTION
**Pull request by NiX3r**
image
image

### CODE REVIEW

File: csharp/send-http-request/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+```
+$ git clone https://github.com/open-runtimes/examples.git && cd examples
+$ cd csharp/send-http-request
+```
+
+2. Enter this function folder and build the code:
+   
+   ```
+   docker run -e INTERNAL_RUNTIME_ENTRYPOINT=src/mod.ts --rm --interactive --tty --volume $PWD:/usr/code openruntimes/deno:1.14 sh /usr/local/src/build.sh
```
[the remark this answers is not in the document]
Comment (NiX3r) [reply csharp/send-http-request/README.md]: Updated. Thanks

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
@NiX3r Thanks for the PR! 🤯 Please give us some time to review it.

**Comment by NiX3r** [comment]
I've done some changes. If you might look at it. Thanks

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
I tried running your code, but got nothing back. here's my request:

```
{"url":"https://demo.appwrite.io/v1/locale/countries/eu","method":"GET","headers":{"x-client-version":"1.0.0"},"body":""}
```

and my result:

```json
{
  "$id": "634dd60d193c85582b43",
  "$createdAt": "2022-10-17T22:24:13.103+00:00",
  "$updatedAt": "2022-10-17T22:24:16.081+00:00",
  "$permissions": [
    "read(\"user:634dcaef4e9b627da03c\")"
  ],
  "functionId": "dotnet-send-http",
  "trigger": "http",
  "status": "completed",
  "statusCode": 200,
  "response": "",
  "stdout": "",
  "stderr": "",
  "duration": 0.267728
}
```

File: csharp/send-http-request/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+   ```
+   docker run -e INTERNAL_RUNTIME_ENTRYPOINT=src/mod.ts --rm --interactive --tty --volume $PWD:/usr/code openruntimes/dotnet:v2-6.0 sh /usr/local/src/build.sh
+   ```
+   
+   As a result, a `code.tar.gz` file will be generated.
+
+3. Start the Open Runtime:
+   
+   ```
+   docker run -p 3000:3000 -e INTERNAL_RUNTIME_ENTRYPOINT=src/mod.ts -e INTERNAL_RUNTIME_KEY=secret-key --rm --interactive --tty --volume $PWD/code.tar.gz:/tmp/code.tar.gz:ro openruntimes/dotnet:v2-6.0 sh /usr/local/src/start.sh
```
[the remark this answers is not in the document]
Comment (gewenyu99) [reply csharp/send-http-request/README.md:79]: Yep :)

File: csharp/send-http-request/Index.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            request.Headers.Add(key, headers[key]);
+        }
+    }
+
+    if(body != null)
+    {
+        request.ContentType = contentType;
+        var streamWriter = new StreamWriter(request.GetRequestStream());
+        streamWriter.Write(body);
+    }
```
Comment (adityaoberai) [inline csharp/send-http-request/Index.cs:65]: Can we please add additional checks in the `if` statement to ensure that if it is a `GET` Request (or other types that don't have a request body), then the `body` and `content type` remain `null`?

In order to debug the empty responses from the functions, I moved the `try` block entry further up and received the response: `Cannot send a content-body with this verb-type.`

image

Removing the `body` and `content_type` properties from the payload solved this issue. However, keeping the `body` as an empty string triggers the error.

**Review by adityaoberai — CHANGES_REQUESTED** [review]
@NiX3r can you please look into the following error as well?

File: csharp/send-http-request/Index.cs (hunk 3 above)
Comment (adityaoberai) [reply csharp/send-http-request/Index.cs:65]:
```suggestion
    if(!String.IsNullOrEmpty(body))
    {
        request.ContentType = contentType;
        var streamWriter = new StreamWriter(request.GetRequestStream());
        streamWriter.Write(body);
    })
    {
        request.ContentType = contentType;
        var streamWriter = new StreamWriter(request.GetRequestStream());
        streamWriter.Write(body);
    }
```

This should do the trick.

File: csharp/send-http-request/Index.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    if (headers != null)
+    {
+        foreach(string key in headers.Keys)
+        {
+            request.Headers.Add(key, headers[key]);
+        }
+    }
+
+    if(body != null)
```
Comment (adityaoberai) [inline csharp/send-http-request/Index.cs:60]:
```suggestion
    if(!String.IsNullOrEmpty(body))
```
This should do the trick

**Review by adityaoberai — CHANGES_REQUESTED** [review]
Suggested some changes @NiX3r

**Comment by Meldiron** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Comment by christyjacob4** [comment]
@NiX3r please address the comments

**Comment by Meldiron** [comment]
Hey there 👋
We are temporarily closing your PR as there has been no activity on it. Feel free to re-open it when you're ready to start working on it again 😇

**Pull request closed without merging**